src/Symfony/Component/DomCrawler/CrawlerInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DomCrawler;
+
+/**
+ * @extends \IteratorAggregate<int, \DOMNode>|\IteratorAggregate<int, \DOM\Node>
+ */
+interface CrawlerInterface extends \Countable, \IteratorAggregate
+{
+    public function add(\DOMNodeList|\DOMNode|array|string|null $node): void;
```

Comment:
isn't this an issue for compat with the new extension ?

Reply:
I understand where the problem is. But I'm not sure to know how to solve this. We still want to have this interface to avoid self-deprecations notices, right? Not sure how to deal with it 🤔